### PR TITLE
Solid Flight Phases and Global Header

### PR DESCRIPTION
**Pull request by zelongzhang**
Implements new solid flight phases according to new state diagrams. 

Convert previous tasks, MonitorForEmergencyShutoff and AbortPhase into gsListenerTask and flightPhaseTask.

Comments out hybrid specific and unused tasks (i.e ParachuteControl, EngineControl).

Adds global definition for rocket type and peripherals in new Global.h file, includes contents of old data.h file.

### CODE REVIEW

File: Core/Src/main.c [hunk 1]
```diff
@@ -361,14 +314,24 @@ int main(void)
...
     osThreadDef(
-        abortPhaseThread,
-        abortPhaseTask,
+        flightPhaseThread,
+		flightPhaseTask,
+        osPriorityHigh,
+        1,
+        configMINIMAL_STACK_SIZE * 3
+    );
+    logDataTaskHandle =
```
Comment (AgamAulakh) [inline Core/Src/main.c]: should this be assigned to the logDataTaskHandle?

File: Core/Src/main.c [hunk 2]
```diff
@@ -361,14 +314,24 @@ int main(void)
...
+    osThreadDef(
+        gsListenThread,
+		gsListenerTask,
         osPriorityHigh,
         1,
         configMINIMAL_STACK_SIZE
     );
-    abortPhaseTaskHandle =
-        osThreadCreate(osThread(abortPhaseThread), NULL);
+    logDataTaskHandle =
```
Comment (AgamAulakh) [inline Core/Src/main.c]: !!

File: Core/Src/main.c (hunk 1 above)
Comment (zelongzhang) [reply Core/Src/main.c]: changed to flightPhaseTaskHandle.

File: Core/Src/main.c (hunk 2 above)
Comment (zelongzhang) [reply Core/Src/main.c]: changed to gsListenTaskHandle.

**Pull request closed without merging**